ldapauthenticator/ldapauthenticator.py

```diff
@@ -300,7 +300,7 @@ def resolve_username(self, username_supplied_by_user):
                 ))
                 user_dn = user_dn[0]
 
-        return user_dn
+        return (user_dn, response[0]["dn"])
```

Comment:
- Is the `["dn"]` attribute guaranteed to exist?

- Does changing the return type of this method count as a breaking change (so this would be released as `2.0.0` instead of `1.3.0`)?

Reply:
As far as I now, there is always a DN, because it is the unique identifier for an LDAP object.

This change would in principle break all users of this function, but I don't think that this function should be considered part of the public API of the authenticator.

Reply:
Based on reading the code, i agree with behrmann, this is an internal function.

Reply:
I wanted to ensure we can trust that we have a `["dn"]` in our response, and I think we do. Why? Because in this documentation, it seems like a search result should always include objectName that has the type of "LDAPDN": https://tools.ietf.org/html/rfc4511#section-4.5.2.

I assume `objectName` in the LDAP specs above is named `dn` when using the ldap3 library. It seems like an plausible assumption to make.